plutus-emulator/src/Control/Monad/Freer/Extras.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE Rank2Types          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE TypeOperators       #-}
+module Control.Monad.Freer.Extras where
+
+import           Control.Lens
+import qualified Control.Monad.Except         as MTL
+import           Control.Monad.Freer
+import           Control.Monad.Freer.Error
+import           Control.Monad.Freer.Internal
+import           Control.Monad.Freer.Reader
+import           Control.Monad.Freer.State
+import           Control.Monad.Freer.Writer
+import qualified Control.Monad.State          as MTL
+
+{- Note [Various raising helpers]
+These are all to help with the issue where you have something of type
+
+Eff effs a
+
+where effs is some *fixed* list of effects. You may then need to insert
+more effects *under* effs to interpret them in terms of. It turns out that
+inserting effects at the *end* of the list is tricky.
```

Comment:
Sorry, I should have written down my other attempt, which was indeed to do that. However, I want to have a single type for the trace right now (otherwise it's *way* more refactoring), so you end up with `data SomeEffects effs a = SomeEffects (forall effs' . Members effs effs' => Eff effs a)` which is a real pain to work with.

Reply:
https://github.com/lexi-lambda/freer-simple/issues/30 describes it a bit more.